layout-server/src/main/java/org/finos/vuu/layoutserver/service/ApplicationLayoutService.java

```diff
@@ -0,0 +1,76 @@
+package org.finos.vuu.layoutserver.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.finos.vuu.layoutserver.model.ApplicationLayout;
+import org.finos.vuu.layoutserver.repository.ApplicationLayoutRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class ApplicationLayoutService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ApplicationLayoutService.class);
+    private static final String DEFAULT_LAYOUT_FILE = "defaultLayout.json";
+    private static ApplicationLayout defaultLayout;
+    private final ApplicationLayoutRepository repository;
+
+    public void createApplicationLayout(String username, JsonNode layoutDefinition) {
+        repository.save(new ApplicationLayout(username, layoutDefinition));
+    }
+
+    public ApplicationLayout getApplicationLayout(String username) {
+        Optional<ApplicationLayout> layout = repository.findById(username);
+
+        if (layout.isEmpty()) {
+            logger.info("No application layout for user, returning default");
```

Comment:
I didn't want to silently return something that wasn't explicitly asked for. It's a legitimate use case for a new user so we don't want to throw an exception*, but it seemed significant enough to acknowledge. I don't have a particular philosophy on logging though.

* Unless we implement another endpoint specifically to get the default layout, then allow the client to call that one if an error code is returned?

Reply:
Yep okay that makes sense. 

> * Unless we implement another endpoint specifically to get the default layout, then allow the client to call that one if an error code is returned?

Hm. I guess this would be an alternative... I don't actually know which way I like more! I'm happy with the one endpoint 🤷‍♀️

Reply:
@vferraro-scottlogic, you're closest to the client-side of this - do you have any input here?

Reply:
I think I like the current approach better than having a separate endpoint for the default layout. That is purely for "keeping it simple" sake though I don't have a string argument against GET defaultLayout being its own endpoint. If we are still on the fence about this maybe we could get Steve's input on it

Reply:
I've stuck with the "log and return default" approach. I've also done a little refactor here - thanks for finding the `orElseGet`, @cfisher-scottlogic!